test/ruby/test_iseq.rb

```diff
@@ -813,6 +813,10 @@ def test_unreachable_syntax_error
   end
 
   def test_unreachable_pattern_matching
+    assert_in_out_err([], "true or 1 in 1")
```

Comment:
Please add a case with more patterns.

```ruby
    assert_in_out_err([], "true or (case 1; in 1; 1; in 2; 2; end)")
```

Reply:
Added in https://github.com/ruby/ruby/pull/11267/commits/1975008c72a75bc007491709fe00ce3010788b06